pallets/eth-bridge/src/tx.rs

```diff
@@ -0,0 +1,107 @@
+use super::*;
+use crate::{util::bound_params, Config};
+use frame_support::BoundedVec;
+use sp_core::Get;
+
+pub fn add_new_request<T: Config>(
+    function_name: &[u8],
+    params: &[(Vec<u8>, Vec<u8>)],
+) -> Result<u32, Error<T>> {
+    let function_name_string =
+        String::from_utf8(function_name.to_vec()).map_err(|_| Error::<T>::FunctionNameError)?;
+    if function_name_string.is_empty() {
+        return Err(Error::<T>::EmptyFunctionName)
+    }
+
+    let tx_id = use_next_tx_id::<T>();
+
+    let tx_request = RequestData {
+        tx_id,
+        function_name: BoundedVec::<u8, FunctionLimit>::try_from(function_name.to_vec())
+            .map_err(|_| Error::<T>::ExceedsFunctionNameLimit)?,
+        params: bound_params(&params.to_vec())?,
+    };
+
+    if ActiveTransaction::<T>::get().is_some() {
+        queue_tx_request(tx_request)?;
+    } else {
+        set_as_active_tx(tx_request)?;
+    }
+
+    Ok(tx_id)
+}
+
+pub fn is_active<T: Config>(tx_id: &u32) -> bool {
+    ActiveTransaction::<T>::get().map_or(false, |active_tx| active_tx.id == *tx_id)
```

Comment:
Why do we have to turn this into a map and do an iterative check? Isn't `ActiveTransaction` a single object and can't we check its `tx_id` immediately?

Reply:
It's not an iterator, it's a function mapping. We need it because ActiveTransaction is an option, since most of the time we won't have one. So all this does is check the id matches (if there is one), otherwise it returns the default value of false (first argument there in the map_or.

I did realise I don't need to pass by ref for a u32 though so I've removed that now.